great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3872,3 +3881,13 @@ def _attach_zep_config_datasources(self, config: GxConfig):
         for ds_name, datasource in config.datasources.items():
             logger.info(f"Loaded '{ds_name}' from ZEP config")
             self._attach_datasource_to_context(datasource)
+
+    def _synchronize_zep_datasources(self) -> Dict[str, XDatasource]:
+        """
+        Update `self.zep_config.xdatasources` with any newly added datasources.
+        Should be called before serializing `zep_config`.
+        """
+        xdatasources = self.xdatasources
+        if xdatasources:
+            self.zep_config.xdatasources.update(xdatasources)
```

Comment:
The other thing I could do right here, is get rid of `__xdatasource_keys` and make the `xdatasources` just do an `isinstance()` check on all the data sources.
Very non-performant, but it would always be correct and handles deleted datasources.

```
  @property
    def xdatasources(self) -> Dict[str, XDatasource]:
        return {
            k: v for k in self.datasources.items() if isinstance(v, XDatasource)
        }
```

Reply:
That sounds simpler. There isn't a huge number of datasources compared to actual data so I'm not concerned with performance here.